pkg/registry/apis/provisioning/register.go

```diff
@@ -1131,28 +1103,26 @@ func (b *APIBuilder) asRepository(ctx context.Context, obj runtime.Object) (repo
 }
 
 func (b *APIBuilder) AsRepository(ctx context.Context, r *provisioning.Repository) (repository.Repository, error) {
+	// Try first with any extra
```

Comment:
this is the change that feels the weirdest -- **BUT**  when I look deeper the wierd part was already there (creating the full webhook flavor repository setup for each request, not just the webhook ones)

Reply:
yeah, still some work there. I think it will become cleaner as we go with other extras. Piece by piece :)